src/reconcile/rec_child.c

```diff
@@ -54,8 +56,21 @@ __rec_child_deleted(
      * setting the page delete structure committed flag cannot overlap with us checking the flag.
      */
     if (__wt_page_del_committed_set(page_del)) {
-        if (F_ISSET(session->txn, WT_TXN_HAS_SNAPSHOT)) {
+        if (F_ISSET(r, WT_REC_VISIBLE_ALL)) {
```

Comment:
Is this the flag that's set for when we are reconciling during a checkpoint?

If so, "VISIBLE ALL" does not seem to be an appropriate name for this flag anymore, and we should rename it (probably in a separate ticket, but either way would work).

Reply:
yes, I agree we should rename it.